File: solutions/uri/1796/1796.cpp
#include <iostream>

int main() {
    int q, v, not_ok_count, survey_count;

    while (std::cin >> q) {
        not_ok_count = 0;
        survey_count = 0;

        while (q--) {
            std::cin >> v;
            if (v) {
                not_ok_count++;
            }
            survey_count++;
        }

        if ((not_ok_count < survey_count / 2 && survey_count % 2 == 0)
            || (not_ok_count < survey_count / 2 + 1 && survey_count % 2 == 1)) {
            std::cout << "Y";
        } else {
            std::cout << "N";
        }
        std::cout << std::endl;
    }

    return 0;
}

File: solutions/beecrowd/1796/1796.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

while line := input().strip():
    q = int(line)
    opinions = map(int, input().split())
    not_ok_count = 0
    survey_count = 0

    for opinion in opinions:
        if opinion:
            not_ok_count += 1
        survey_count += 1

    if (not_ok_count < survey_count // 2 and survey_count % 2 == 0) or (
        not_ok_count < survey_count // 2 + 1 and survey_count % 2 == 1
    ):
        output('Y\n')
    else:
        output('N\n')
